src/main/groovy/io/openliberty/tools/gradle/tasks/AbstractFeatureTask.groovy

```diff
@@ -185,8 +204,26 @@ public class AbstractFeatureTask extends AbstractServerTask {
         }
     }
 
-    protected InstallFeatureUtil getInstallFeatureUtil(Set<String> pluginListedEsas, List<ProductProperties> propertiesList, String openLibertyVerion, String containerName) throws PluginExecutionException {
-        createNewInstallFeatureUtil(pluginListedEsas, propertiesList, openLibertyVerion, containerName)
+    protected InstallFeatureUtil getInstallFeatureUtil(Set<String> pluginListedEsas, List<ProductProperties> propertiesList, String openLibertyVerion, String containerName, List<String> additionalJsons) throws PluginExecutionException {
+		//if installing userFeature, recompile gradle to find mavenLocal artifacts created by prepareFeature task. 
+		if(project.configurations.featuresBom.dependencies) {  
+			try {
+				ProjectBuilder builder = ProjectBuilder.builder();
+				newProject = builder
+						.withProjectDir(project.rootDir)
+						.withGradleUserHomeDir(project.gradle.gradleUserHomeDir)
+						.withName(project.name)
+						.build();
+
+				// need this for gradle to evaluate the project
+				// and load the different plugins and extensions
+				newProject.evaluate();
+			} catch (Exception e) {
+				logger.error("Could not parse build.gradle " + e.getMessage());
+				logger.debug('Error parsing build.gradle', e);
+			}
+		}
+		createNewInstallFeatureUtil(pluginListedEsas, propertiesList, openLibertyVerion, containerName, additionalJsons)
```

Comment:
If the above section fails (gets an exception), we are allowing it to continue on? Will it likely fail?

Reply:
Added return statement when the exception is caught to make it fail.

Reply:
If you are going to return, shouldn't it `return null` ?

Reply:
Actually, the caller does not check the return value and tries to call a method on it. I think the exception needs to be propagated rather than eaten here.

Reply:
Yeah I was just going over that, I will make it throw PluginExecutionException instead.